### ISSUE TITLE
Cannot set participant data - Serialization Issue

### ISSUE THREAD

**Issue by bardram** [cph-cachet/carp-webservices-spring#114]
When calling "SetParticipantData" with one of the new non-core input type (Address) I get an error.

Request:

```
{
 "__type": "dk.cachet.carp.deployments.infrastructure.ParticipationServiceRequest.SetParticipantData",
 "studyDeploymentId": "643e3c37-f1de-46af-896f-0a2b2b46bc92",
 "apiVersion": "1.0",
 "data": {
  "dk.carp.webservices.input.address": {
   "__type": "dk.carp.webservices.input.address",
   "address1": "Ørsteds Plads",
   "city": "Kgs. Lyngby",
   "postalCode": "DK-2800",
   "country": "Denmark"
  }
 }
}
```

Response:

```
{"statusCode":400,"exception":"dk.cachet.carp.webservices.common.exception.serialization.SerializationException","message":"The ParticipationServiceRequest deserializer is not valid. Exception: Fields [address2, street] are required for type with serial name 'dk.carp.webservices.input.address', but they were missing at path: $.data['dk.carp.webservices.input.address']","path":"POST /api/participation-service"}
``` 

All of the fields in the address should be nullable - right?

**Comment by bardram** [comment]
Now - trying to add all field, I have this request:

```
      {
       "__type": "dk.cachet.carp.deployments.infrastructure.ParticipationServiceRequest.SetParticipantData",
       "studyDeploymentId": "643e3c37-f1de-46af-896f-0a2b2b46bc92",
       "apiVersion": "1.0",
       "data": {
        "dk.carp.webservices.input.address": {
         "__type": "dk.carp.webservices.input.address",
         "address1": "DTU HealthTech",
         "address2": "Technical University of Denmark",
         "street": "Ørsteds Plads",
         "city": "Kgs. Lyngby",
         "postalCode": "DK-2800",
         "country": "Denmark"
        }
       }
      }
```

But this ALSO fails with this response:

```
 {"statusCode":400,"exception":"java.lang.IllegalArgumentException","message":"The input data type is not assigned to this participant role.","path":"POST /api/participation-service"}
```

**Comment by bardram** [comment]
When setting the Informed Consent, I get this error:

```
[log] [CAMS DEBUG] RESPONSE: 400
      {"statusCode":400,"exception":"dk.cachet.carp.webservices.common.exception.serialization.SerializationException","message":"The ParticipationServiceRequest deserializer is not valid. Exception: Fields [userId, consent, signatureImage] are required for type with serial name 'dk.carp.webservices.input.informed_consent', but they were missing at path: $.data['dk.carp.webservices.input.informed_consent']","path":"POST /api/participation-service"}
```

which seems to indicate that the fields `userId`, `consent`, and `signatureImage`  are required -- However, that does not allign with the [documentation](https://github.com/cph-cachet/carp-webservices-spring/blob/develop/docs/input-data-types.md#informedconsent).

**Comment by bardram** [comment]
So - in general, there seems to be a mismatch between the [documentation](https://github.com/cph-cachet/carp-webservices-spring/blob/develop/docs/input-data-types.md) and the actual code....

**Comment by bardram** [comment]
Same problem with `FullName`.

```
CarpServiceException: 400 Bad Request -  The ParticipationServiceRequest deserializer is not valid. Exception: Fields [firstName, middleName, lastName] are required for type with serial name 'dk.carp.webservices.input.full_name', but they were missing at path: $.data['dk.carp.webservices.input.full_name'] - POST /api/participation-service
package:carp_webservices/carp_services/carp_base_service.dart 113:5  CarpBaseService._rpc
```

**Comment by yuanchen233** [comment]
Yes, the value to set can be null, but the field is needed in the request body for the serializer to work.
This behavior can be changed by adding a default value in the backend.

**Comment by yuanchen233** [comment]
Fixed #115
